rmw_fastrtps_shared_cpp/src/rmw_publisher.cpp

```diff
@@ -119,6 +121,35 @@ __rmw_publisher_assert_liveliness(
   return RMW_RET_OK;
 }
 
+rmw_ret_t
+__rmw_publisher_wait_for_all_acked(
+  const char * identifier,
+  const rmw_publisher_t * publisher,
+  rmw_time_t wait_timeout)
+{
+  RMW_CHECK_ARGUMENT_FOR_NULL(publisher, RMW_RET_INVALID_ARGUMENT);
+  RMW_CHECK_TYPE_IDENTIFIERS_MATCH(
+    publisher,
+    publisher->implementation_identifier,
+    identifier,
+    return RMW_RET_INCORRECT_RMW_IMPLEMENTATION);
+
+  auto info = static_cast<CustomPublisherInfo *>(publisher->data);
+  if (nullptr == info) {
+    RMW_SET_ERROR_MSG("publisher internal data is invalid");
+    return RMW_RET_ERROR;
+  }
```

Comment:
I've been told by other reviewers to avoid this kind of check. If this pointer is null it is either because the user of the API broke the contract, or something was internally broken.

```suggestion
```

Reply:
Get it.